idl/storage_proxy.idl.hh

```diff
@@ -26,7 +26,7 @@
 verb [[with_client_info, with_timeout, one_way]] mutation (frozen_mutation fm [[ref]], inet_address_vector_replica_set forward [[ref]], gms::inet_address reply_to, unsigned shard, uint64_t response_id, std::optional<tracing::trace_info> trace_info [[ref]] [[version 1.3.0]], db::per_partition_rate_limit::info rate_limit_info [[version 5.1.0]], service::fencing_token fence [[version 5.4.0]]);
 verb [[with_client_info, one_way]] mutation_done (unsigned shard, uint64_t response_id, db::view::update_backlog backlog [[version 3.1.0]]);
 verb [[with_client_info, one_way]] mutation_failed (unsigned shard, uint64_t response_id, size_t num_failed, db::view::update_backlog backlog [[version 3.1.0]], replica::exception_variant exception [[version 5.1.0]]);
-verb [[with_client_info, with_timeout]] counter_mutation (std::vector<frozen_mutation> fms, db::consistency_level cl, std::optional<tracing::trace_info> trace_info [[ref]]);
+verb [[with_client_info, with_timeout]] counter_mutation (std::vector<frozen_mutation> fms, db::consistency_level cl, std::optional<tracing::trace_info> trace_info [[ref]]) -> replica::exception_variant [[version 5.4.0]];
```

Comment:
> I thought that [[version ]] tag is only for documentation, and doesn't affect RPC behavior in any way.

The `[[version]]` makes type type `rpc::optional`, so that the new node can check if the data was sent by the old replica.

> What happens if an old node which works with the old definition gets a non-empty result, is it ignored?

I looked at the code, we read the entire frame in `seastar::rpc::connection::read_frame`, then deserialize the items in order of declaration (`do_unmarshall`), so as long as we don't change the order of items and keep adding them at the end we should be fine.